mono/mini/interp/interp.c

```diff
@@ -2520,16 +2521,27 @@ interp_exec_method_full (InterpFrame *frame, ThreadContext *context, guint16 *st
 					goto exit_frame;
 			}
 			ip += 2;
-			if (new_method->alloca_size > rtm->alloca_size)
-				g_error ("MINT_JMP to method which needs more stack space (%d > %d)", new_method->alloca_size, rtm->alloca_size); 
 			rtm = frame->imethod = new_method;
+			/*
+			 * We allocate the stack frame from scratch and store the arguments in the
+			 * locals again since it's possible for the caller stack frame to be smaller
+			 * than the callee stack frame (at the interp level)
+			 */
+			if (realloc_frame) {
+				frame->args = alloca (rtm->alloca_size);
+				memset (frame->args, 0, rtm->alloca_size);
```

Comment:
shouldn't this use `new_method->alloca_size`?

Reply:
the comment doesn't match what the code is doing, right? it doesn't copy the arguments over

Reply:
At this point, rtm is the new_method.
The arg copying should be done by the stinargs at the beginning of the method (we set the ip to the start of the method below)

Reply:
ah I see, now it makes sense 🙂